File: java/src/main/java/org/example/crypto/HashingExample.java
package org.example.crypto;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;

public final class HashingExample {
    private HashingExample() {}

    public static byte[] sha256(byte[] data) throws Exception {
        return MessageDigest.getInstance("SHA-256").digest(data);
    }

    public static void main(String[] args) throws Exception {
        byte[] digest = sha256("secure coding reference".getBytes(StandardCharsets.UTF_8));
        System.out.println("HashingExample SHA-256=" + CryptoSupport.hex(digest));
    }
}

File: python/hashing_example.py
"""SHA-256 hashing example. Hashes are not password storage or message authentication."""

from cryptography.hazmat.primitives import hashes


def sha256(data: bytes) -> bytes:
    digest = hashes.Hash(hashes.SHA256())
    digest.update(data)
    return digest.finalize()


def main() -> None:
    print(f"HashingExample SHA-256={sha256(b'secure coding reference').hex()}")


if __name__ == "__main__":
    main()
